### PR TITLE
Fix bug to prevent display of participants that have no edge to study

### PR DESCRIPTION
**Pull request by giuliacampana**

### CODE REVIEW

File: src/containers/study/StudySagas.js [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 /* eslint-disable no-use-before-define */
 
+import _ from 'lodash';
```
Comment (UnsungHero97) [inline src/containers/study/StudySagas.js]: Don't do this. You can import any lodash function directly:

```
import uniqBy from 'lodash/uniqBy';
```

File: src/containers/study/StudySagas.js [hunk 2]
```diff
@@ -169,19 +170,47 @@ function* getParticipantsWorker(action :SequenceAction) :Generator<*, *, *> {
...
+      },
+    };
+    response = yield call(
+      searchEntityNeighborsWithFilterWorker,
+      searchEntityNeighborsWithFilter(params),
+    );
+    // NOTE: sometimes, a participant will have more than one edge to the study
+    let filteredParticipants = [];
+    if (Object.keys(response.data).length) {
+      filteredParticipants = _.uniqBy(response.data[studyEntityKeyId], neighbor => neighbor.neighborId);
```
Comment (UnsungHero97) [inline src/containers/study/StudySagas.js]: Any reason why you're not using Immutable?

File: src/containers/study/StudySagas.js [hunk 3]
```diff
@@ -169,19 +170,47 @@ function* getParticipantsWorker(action :SequenceAction) :Generator<*, *, *> {
...
+        entityKeyIds: [studyEntityKeyId],
+      },
+    };
+    response = yield call(
+      searchEntityNeighborsWithFilterWorker,
+      searchEntityNeighborsWithFilter(params),
+    );
+    // NOTE: sometimes, a participant will have more than one edge to the study
+    let filteredParticipants = [];
+    if (Object.keys(response.data).length) {
```
Comment (UnsungHero97) [inline src/containers/study/StudySagas.js]: This will throw if `response.data` is not there.

File: src/containers/study/StudySagas.js [hunk 4]
```diff
@@ -169,19 +170,47 @@ function* getParticipantsWorker(action :SequenceAction) :Generator<*, *, *> {
...
+      entitySetId: studiesEntitySetId,
+      filter: {
+        edgeEntitySetIds: [participatedInEntitySetId],
+        entityKeyIds: [studyEntityKeyId],
+      },
+    };
+    response = yield call(
+      searchEntityNeighborsWithFilterWorker,
+      searchEntityNeighborsWithFilter(params),
+    );
```
Comment (UnsungHero97) [inline src/containers/study/StudySagas.js]: You should check for error:

```
if (response.error) {
  throw response.error;
}
```

File: src/containers/study/StudySagas.js [hunk 5]
```diff
@@ -321,36 +351,44 @@ function* getParticipantEnrollmentStatusWorker(action :SequenceAction) :Generato
...
+        searchEntityNeighborsWithFilter(params),
+      );
+      if (response.error) throw response.error;
+
+      const participantNeighborResults :Map = fromJS(response.data);
+
+      // NOTE: participant may have multiple edges but status should be the same on all edges, so grab first edge
+      participantEnrollmentStatus = participantNeighborResults
+        .map((participant :List) => {
+          if (participant.count() > 1) {
```
Comment (UnsungHero97) [inline src/containers/study/StudySagas.js]: Why is this being reverted?

File: src/containers/study/StudySagas.js (hunk 1 above)
Comment (UnsungHero97) [reply src/containers/study/StudySagas.js]: ESLint should also complain about the `_`.

File: src/containers/study/StudySagas.js (hunk 5 above)
Comment (giuliacampana) [reply src/containers/study/StudySagas.js]: That's a mistake—thanks for catching it.

File: src/containers/study/StudySagas.js (hunk 1 above)
Comment (giuliacampana) [reply src/containers/study/StudySagas.js]: Okay thanks.

File: src/containers/study/StudySagas.js (hunk 2 above)
Comment (giuliacampana) [reply src/containers/study/StudySagas.js]: I couldn't find an equivalent function for Lists in Immutable to the uniqBy function from `lodash`.

**Pull request merged**